src/Symfony/Component/Validator/Violation/ConstraintViolationBuilder.php

```diff
@@ -141,7 +142,7 @@ public function setCause($cause)
      */
     public function addViolation()
     {
-        if (null === $this->plural) {
+        if (null === $this->plural || !$this->translator instanceof LegacyTranslatorInterface) {
```

Comment:
This will cause some deprecations. How can I know if I should use transChoice or not?